cm_sdk/models/__init__.py

```diff
@@ -0,0 +1,266 @@
+        return self.state.keys().__iter__()
+
+    def __setattr__(self, name, value):
+
+        if name not in self.all_api_attributes:
+            raise CloudManagerValidationException("Attribute {0} is not allowable for type {1}".
+                                                  format(name, type(self).__name__))
+
+        # Verify that the name in the JSON corresponds to list of children defined by each model.
+        # If it is, we dispatch here to get an instance of the appropriate object or enum
+        if name in self.children:
+
+            if isinstance(value, list):
+                value = self.get_children(name, value)
+            else:
+                value = self.get_child(name, value)
+
+        # Do not store nil values
+        if value:
+            self.state[name] = value
+
+    def get_child(self, name, value):
+        if isinstance(value, (CloudManagerEnum, CloudManagerBase)):
+            return value
+        else:
+            if value is None:
+                return value
+            else:
+                return self.get_child_from_value(name, value)
+
+    def get_child_from_value(self, name, value):
+
+        if hasattr(self, "children") \
+                and name in self.children:
+            try:
+                klass = self.children[name]['class']
+
+                if issubclass(klass, self.get_class_from_name('cm_sdk.models.CloudManagerEnum')):
+                    try:
+                        value = klass[value]
+                    except KeyError:
+                        logging.warn("Enum {0} has no key {1}".format(klass, value))
+                elif issubclass(klass, self.get_class_from_name('cm_sdk.models.CloudManagerBase')):
+                    value = klass.from_dict(value)
+                else:
+                    raise CloudManagerValidationException(
+                        "Cannot marshall class {0} in deserializing from API JSON".format(klass.__class__))
+
+            except AttributeError:
+                print "Problem with {name}".format(name=name)
+
+        return value
+
+    def get_children(self, name, items):
+
+        to_return = []
+
+        for item in items:
+            if isinstance(item, CloudManagerBase) or isinstance(item, CloudManagerEnum):
+                to_return.append(item)
+            else:
+                to_return.append(self.get_child(name, item))
+
+        return to_return
+
+    @classmethod
+    def from_json(cls, json_in):
+        _dict = json.loads(json_in)
+        return cls.from_dict(StringUtil.convert_dict_keys(_dict, StringUtil.Case.SNAKE))
+
+    @classmethod
+    def from_dict(cls, dict_in):
+        instance = cls()
+        converted = StringUtil.convert_dict_keys(dict_in, StringUtil.Case.SNAKE)
+
+        for k, v in converted.items():
+            setattr(instance, k, v)
+        return instance
+
+    def to_json(self):
+        return json.dumps(self, cls=CloudManagerJSONEncoder)
+
+
+class OrderedEnum(Enum):
```

Comment:
Where do you use the properties granted by `OrderedEnum`? If all you need is to use `in`, you can get the same thing by using `Enum.__members__`.